cpp/src/arrow/parquet/writer.h

```diff
@@ -52,6 +54,14 @@ class FileWriter {
   std::unique_ptr<Impl> impl_;
 };
 
+/**
+ * Write a flat Table to Parquet.
+ *
+ * The table shall only consist of nullable, non-repeated columns of primitive type.
```

Comment:
Thanks -- I misread the comment (this is the write path). It would be clearer to say that the written Parquet columns will all be of OPTIONAL RepetitionType, but this is OK for now.